metronome/hotstuff/service/src/io/iohk/metronome/hotstuff/service/SyncService.scala

```diff
@@ -253,6 +254,25 @@ class SyncService[F[_]: Concurrent: ContextShift, N, A <: Agreement: Block](
         .void
   }
 
+  /** Validate the prepared block after the parent has been downloaded. */
+  private def validateBlock(block: A#Block): OptionT[F, Boolean] =
```

Comment:
I realise this makes the code slightly more concise, but I tend to think of monad transformers as tools to be used in-place rather than passed around.

Not a blocker, but I'm curious about your opinion.

Reply:
I only did it this way because I didn't like to do this:
```scala
def validate = 
  (for {
    a <- OptionT(foo)
    b <- OptionT(bar)
  } yield a+b).value
```
or this:
```scala
def validate = { 
  val howDoINameThis = for {
    a <- OptionT(foo)
    b <- OptionT(bar)
  } yield a+b

  howDoINameThis.value
}
```

I also think that if there are too many `OptionT` values in the expression it's actually better to pass them around, and only take the `.value` at the very end, rather than having to keep rewrapping them. The goal IMO is to make things readable, to make the extra syntax dealing with `Option` disappear. 

I remember that it was really confusing to see all the `lift` and `EitherT` everywhere in a massive `for` expression, when I first saw them being applied, it obscured the logic. 

Another reason I thought this could make a difference is that both methods are just called `validateBlock`, so I thought it lowers the chance of mixing up `appService.validateBlock` with this if they have a different return type.

I don't have a strong opinion in this case, it just felt less awkward.

Reply:
Removed `OptionT` in https://github.com/input-output-hk/metronome/pull/55/commits/57d08834ba4c9d041036f1d468fd4d085dc997e3